src/SourceLocator/Type/DirectoriesSourceLocator.php

```diff
@@ -28,10 +29,10 @@ class DirectoriesSourceLocator implements SourceLocator
      * @throws InvalidDirectory
      * @throws InvalidFileInfo
      */
-    public function __construct(array $directories)
+    public function __construct(array $directories, ?Locator $locator = null)
```

Comment:
If we're adding params, let's not be afraid to break BC. This is mostly internal stuff, so I think it's probably OK to not have this nullable